js/app.js

```diff
@@ -1 +1,92 @@
-console.log('Hello frontend');
+ var player1;
+ var player2;
+ var playerTurn;
+
+ var xButton = document.getElementById("js_x");
+ var oButton = document.getElementById("js_o");
+
+ xButton.addEventListener("click", function(event, once) {
+ 	if (player1) {
+ 		return
+ 	} else {
+ 		changeColor(event.currentTarget);
+ 		assignPlayer(event.currentTarget);
+ 		playGame();
+ 	}
+ });
+ oButton.addEventListener("click", function(event, once) {
+ 	if (player2) {
+ 		return
+ 	} else {
+ 		changeColor(event.currentTarget);
+ 		assignPlayer(event.currentTarget);
+ 		playGame();
+ 	}
+ });
+
+ function changeColor(button){
+ 	if (button == xButton) {
+ 		xButton.style.backgroundColor = "purple";
+ 		oButton.style.backgroundColor = "";
+ 	} else if (button === oButton) {
+ 		oButton.style.backgroundColor = "purple";
+ 		xButton.style.backgroundColor = "";
+ 	}
+ }
+
+function assignPlayer(input) {
+	if (input == xButton) {
+		player1 = "X";
+		player2 = "O";
+	} else if (input == oButton) {
+		player1 = "O";
+		player2 = "X";
+	} else {
+		return;
+	}
+	playerTurn = player1;
+}
+
+var gameboard = {
+	row1: {
+		leftCell: document.getElementById("js_1"),
+		centerCell: document.getElementById("js_2"),
+		rightCell: document.getElementById("js_3")
+	},
+	row2: {
+		leftCell: document.getElementById("js_4"),
+		centerCell: document.getElementById("js_5"),
+		rightCell: document.getElementById("js_6")
+	},
+	row3: {
+		leftCell: document.getElementById("js_7"),
+		centerCell: document.getElementById("js_8"),
+		rightCell: document.getElementById("js_9")
+	}
```

Comment:
Imagine you were instead of tic-tac-toe now implementing connect-4. You have to get 4 in a row and you now have a 7x6 board space. Would the way you defined the gameboard be different or the same? Why/Why not?